store/db/migration/prod/LATEST__SCHEMA.sql

```diff
@@ -137,3 +137,20 @@ CREATE TABLE memo_relation (
   type TEXT NOT NULL,
   UNIQUE(memo_id, related_memo_id, type)
 );
+
+-- memo_comment
+CREATE TABLE memo_comment (
+  id INTEGER PRIMARY KEY AUTOINCREMENT,
+  created_ts BIGINT NOT NULL DEFAULT (strftime('%s', 'now')),
+  updated_ts BIGINT NOT NULL DEFAULT (strftime('%s', 'now')),
+  content TEXT NOT NULL DEFAULT '',
+  email TEXT NOT NULL DEFAULT '',
+  website TEXT DEFAULT '',
+  name TEXT NOT NULL DEFAULT 'UNKNOWN',
+  memo_id INTEGER NOT NULL,
+  parent_id INTEGER DEFAULT (0)
```

Comment:
For the first version, I'd like to remove the `website` and `parent_id`. Let's keep it as simple as possible for now. And we can always add it back later if we really need it. 

```suggestion
  email TEXT NOT NULL DEFAULT '',
  website TEXT DEFAULT '',
  name TEXT NOT NULL DEFAULT 'UNKNOWN',
  memo_id INTEGER NOT NULL,
  parent_id INTEGER DEFAULT (0)
```

Reply:
我来解释一下这几个字段为什么是必须的。

email和website：以下我的思考和设计是参考很多的博客评论功能。memos大部分都是博客圈的人在用，而博客圈也好、微博也好，留言的较大动力和社交关系的建立，都主要是通过留言、回访来达成，如果去掉了email和website，那么类似于失去了灵魂。同时，增加这两个字段也是让发布评论的人，对自己说的话、编写的字更有责任感、更斟酌一些。

parent_id：是必须的，这个parent_id用来形成对于某条评论的回复，代表回复的层级关系，是一个树形设计。评论回复是增加在评论区的交流机会，这并不是多余字段的。

如果失去了这些，那么就变成了单方面访客的点评系统了，这违背了设计评论系统的初衷。

所以我建议尽可能还是保留这几个字段，我们减少了这几个字段，并不能给系统带来更好的体验，那为什么咱们要在第一版本中删除掉呢，对吗？

Reply:
我把Memos当self-host微博使用，很希望有个原生评论，所以一直在关注这个功能，感谢各位大佬的贡献。说下看法：

email + parent_id 基本是评论功能的MVP了，为回复和通知能力做准备。如果没有这个能力，评论功能确实没什么意义。
website倒是可选，各个博客程序也是如此设计。就我个人而言，希望保留，我经常通过评论留言的网站去大家的个人博客逛逛，有社交属性。

如果评论功能完成了，email通知功能也要开始考虑了，否则少了很多乐趣。

Reply:
* 首先这个是匿名评论，存的这些信息有可能都是假的垃圾信息，攻击者可以随意通过 api 来 create comment。
* 创建者相关的信息如果每个都用一个字段来存储，之后如果要加一个信息（比如 GitHub 用户名），那就是需要再加一个字段？这里我觉得可以把这些信息放到一个字段里，存为 JSON string，展示的时候再提取出来。
* 至于 `parent_id`，第一版我倾向于先不做嵌套评论。

Reply:
- 目前我所见过的不需要登录的匿名评论，都可能会存在假的垃圾信息，匿名与信息的质量之间，我们无法做到完美，暂时只能做到平衡。所以，既然是一个开放生态，要呈现一个匿名评论系统，就不能过多考虑垃圾信息这个问题，就现有的各类如artalk、twikoo等已经非常优秀，但目前也只是通过 各类过滤 来实现垃圾屏蔽。
- 我觉得boojack 老师第二点我并不是很认同，但我能理解。不认同的原因是，**评论**，并不单单是一段文字，而是一个复杂的结合体，例如，一辆汽车发动机是核心，但如果只有发动机，那就只是一个展台上的物件，评论是由 人（是谁）、来自哪里（website）、说了什么（content）、以及我如何回应他（Email）. 根据数据库设计范式，我们理论上是增加一类信息，就要增加一个字段的，我们很多时候容易陷入一个少加字段，字段不方便扩展等等误区。想想，字段被发明出来的意义不就是组合、呈现、扩展吗？
- parent_id ，emmm，哈哈哈我确实感觉没有嵌套（这里的嵌套不是单单只界面上的嵌套展示），就没有交流感，没有交流感，评论的意义在哪里？囧 

当然，boojack老师的严谨的想法没有问题，只是我们没有达成共识，那么按照创造者优先原则，按照你说的来做吧。

Reply:
不同的使用场景考虑点可能不一定一样。

我觉得首先先实现一个最小系统，能够拿到一些反馈，再据此来调整。那样的话，增加、完善某项功能，还是移除某些特性甚至去掉整个功能，都更合适一些。